src/Promise.php

```diff
@@ -243,7 +243,7 @@ private function invokeWaitFn()
         try {
             $wfn = $this->waitFn;
             $this->waitFn = null;
-            $wfn(true);
+            $wfn($this);
```

Comment:
Ah I see. It's related to https://github.com/guzzle/promises/issues/57. Is this change necessary for this PR? If this isn't needed for this PR, then I'd prefer these PRs to be broken into two different PRs.

I don't see anywhere in the docs that the wait function is expecting a boolean argument or that it is being used anywhere. I think this change and #57 make sense. We should probably update the documentation and docblocks to show this change as well.